### ISSUE TITLE
Trouble updating Vendor Custom Fields

### ISSUE THREAD

**Issue by bcpouli** [Intacct/intacct-sdk-net#74]
I'm using the C# SDK object VendorUpdate to update some vendor data, all is well until I try to update some custom fields that were added to the vendors.

When I try to save values to the custom fields using this code:

```
            VendorUpdate update = new VendorUpdate()
            {
                VendorId ="VendorIDHere"
            };

             update.CustomFields["FIELD_NAME"] = "data";
             Task<OnlineResponse> updateTask = client.Execute(update);
             updateTask.Wait();
```

I get this error:

Microsoft.CSharp.RuntimeBinder.RuntimeBinderException: The call is ambiguous between the following methods or properties: 'Intacct.SDK.Xml.IaXmlWriter.WriteElement(string, decimal?, bool)' and 'Intacct.SDK.Xml.IaXmlWriter.WriteElement(string, bool?, bool)'

However, if I go into the Intacct front-end and update any custom field by hand first, then run the update through the SDK, I don't get any errors and the field updates correctly. 

I then tried to perform this operation using PostMan and just using the API to ensure that it wasn't an API issue.  I used this XML in my call:

```
<update>
	<VENDOR>
		<VENDORID>VENDORIDHERE</VENDORID>
	        <customfields>
			<customfield> 
                                <customfieldname>CUSTOM_FIELD_NAME</customfieldname>
				<customfieldvalue>VAL</customfieldvalue>
			</customfield>
		</customfields>
	</VENDOR>
</update>
```
This updated the custom field appropriately for vendors that had not been updated before.  Therefore, it seems this issue only lies with the SDK.

Please let me know if I'm just misunderstanding or misusing the SDK.

**Comment by jimmymcpeter** [comment]
That xml block above with the nested `customfields` element should have only worked in a `update_vendor` function, not `update`.

You might need to use a dictionary like the test is here - https://github.com/Intacct/intacct-sdk-net/blob/4da743c40cb32fa20ac68631180d3e70f38f6aed/Intacct.SDK.Tests/Functions/AccountsPayable/VendorUpdateTest.cs#L205-L209

**Comment by bcpouli** [comment]
Thanks for your response.  After investigating a bit more I found that my issue was being caused by passing null to the call.  This could occur because we were accepting inputs from a screen for users to update their data on, so if they didn't enter anything for some custom fields, it would send null, which caused an error to occur. I changed my code like this:

update.CustomFields["CUSTOM_FIELD"] = obj.Val== null ? "" : obj.Val;

And now I'm not longer having the issue.

Thanks for your help!
